### PR TITLE
Mobile Responsivity of ProductKit Component

### PR DESCRIPTION
**Pull request by gugabribeiro**
#### What is the purpose of this pull request?
Add media queries to treat the behavior of the component when it in mobile screens.

#### What problem is this solving?
Some informations being not visible when the component was rendered in a mobile screen.

#### Screenshots or example usage
<a href="https://gustavo--storecomponents.myvtex.com/ninja-300/p">Workspace</a>

captura de tela de 2018-07-16 17-46-46

captura de tela de 2018-07-16 17-47-09

#### Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/ProductKit.js [hunk 1]
```diff
@@ -85,40 +127,51 @@ class ProductKit extends Component {
     } = this.props
 
     if (!product.benefits.length) {
-      return ( <Fragment></Fragment> )
+      return ( 
+        <Fragment></Fragment>
```
Comment (ClaudivanFilho) [inline react/ProductKit.js]: `return null` isn't better?

File: react/ProductKit.js [hunk 2]
```diff
@@ -65,11 +65,53 @@ class ProductKit extends Component {
...
+            Price: 0,
+            ListPrice: 0,
+          },
+        }
+      }
+      if (newProduct.sku.images && newProduct.sku.images.length) {
+        newProduct.sku.image = { ...newProduct.sku.images[0] }
+        newProduct.sku.image.imageUrl = newProduct.sku.image.imageUrl
+          .replace('http:', '')
+          .replace('https:', '')
```
Comment (lucasecdb) [inline react/ProductKit.js]: Use this regex instead `/^https?:/`

File: react/ProductKit.js [hunk 3]
```diff
@@ -65,11 +65,53 @@ class ProductKit extends Component {
...
+  /**
+   * Calculates the Kit Price according to the products that are being displayed. 
+   * The price itself is calculated based on the discount that each product have separately.
+   */
+  calculatePrice = kitProducts => {
     let kitPrice = 0
-    kitItems.slice(0, DEFAULT_MAX_ITEMS).map(kitItem => {
-      const { discount, product: { items: [ { sellers: [ { commertialOffer: { Price } } ] } ] } } = kitItem
-      kitPrice += Price * (100.0 - discount) / 100.0
+    kitProducts.map(kitProduct => {
```
Comment (lucasecdb) [inline react/ProductKit.js]: Use `reduce` instead of `map`

File: react/ProductKit.js (hunk 1 above)
Comment (lucasecdb) [inline react/ProductKit.js]: You can just return `null`

File: react/ProductKit.js [hunk 4]
```diff
@@ -85,40 +127,51 @@ class ProductKit extends Component {
...
+      return ( 
+        <Fragment></Fragment>
+      )
     }
 
-    const { benefits: [ benefit] } = product
-    const kitItems = benefit.items
+    const { benefits: [ benefit ] } = product
+    
+    const kitProducts = benefit.items.slice(0, MAX_ITEMS).map(item => (
```
Comment (lucasecdb) [inline react/ProductKit.js]: When the function has the same parameters as the callback called by `map`, you can pass the reference to the function directly, e.g. `array.map(this.prepareProduct)`

File: react/ProductKit.js [hunk 5]
```diff
@@ -85,40 +127,51 @@ class ProductKit extends Component {
...
+                  product={kitProduct}
+                  summaryProps={
+                    { 
+                      showListPrice, 
+                      showLabels, 
+                      showInstallments,
+                      showBadge, 
+                      badgeText,
+                    }
+                  } />
```
Comment (lucasecdb) [inline react/ProductKit.js]: Put `/>` in a new line :pick:

File: react/ProductKit.js [hunk 6]
```diff
@@ -65,11 +65,53 @@ class ProductKit extends Component {
...
+          commertialOffer: {
+            Price: 0,
+            ListPrice: 0,
+          },
+        }
+      }
+      if (newProduct.sku.images && newProduct.sku.images.length) {
+        newProduct.sku.image = { ...newProduct.sku.images[0] }
+        newProduct.sku.image.imageUrl = newProduct.sku.image.imageUrl
+          .replace('http:', '')
```
Comment (ClaudivanFilho) [inline react/ProductKit.js]: .replace(/^(http(s?):)/, '')

**Pull request merged**